### ISSUE TITLE
Spotty CoffeeScript detection

### ISSUE THREAD

**Issue by joepie91** [knex/knex#604]
I have a buildscript that compiles a bunch of CoffeeScript files to Javascript files as soon as they change (I simply use the CS files as source files), and one of these files is the knexfile. The goal of this is to not require CoffeeScript to be present or used in production. My migration files are still in CoffeeScript.

However, after adding `knexfile.coffee` to my list of files to be compiled, knex started detecting my knex setup as being Javascript-based instead of CoffeeScript-based - now, every time I run a `migrate:make` command it will give me a Javascript template instead of a CoffeeScript template, and every time I run `migrate:latest` it will try to run my CoffeeScript files in Node.js directly and fail.

Basically, the current logic for language detection appears to be "if Javascript Knex files exist, use JS, and only if they don't use CoffeeScript", and it would probably be more logical to have it the other way around, due to buildscripts like this - that is, to have it default to CoffeeScript if there are any Knex-related CoffeeScript files in the repository, and only assume Javascript if there aren't.

A temporary workaround appears to be to explicitly specify `--knexfile knexfile.coffee`, but it's not very practical to have to specify this for each command individually. For clarification, the `knexfile` needs to be compiled into Javascript because the application itself also uses it for its database configuration.
